verify.go

```diff
@@ -0,0 +1,120 @@
+package merkletree
+
+import (
+	"bytes"
+	"hash"
+)
+
+// VerifyProof takes a Merkle root, a proofSet, and a proofIndex and returns
+// true if the first element of the proof set is a leaf of data in the Merkle
+// root. False is returned if the proof set or Merkle root is nil, and if
+// 'numLeaves' equals 0.
+func VerifyProof(h hash.Hash, merkleRoot []byte, proofSet [][]byte, proofIndex uint64, numLeaves uint64) bool {
+	// Return false for nonsense input. A switch statement is used so that the
+	// cover tool will reveal if a case is not covered by the test suite. This
+	// would not be possible using a single if statement due to the limitations
+	// of the cover tool.
+	switch {
+	case merkleRoot == nil:
+		return false
+	case proofIndex >= numLeaves:
+		return false
+	}
```

Comment:
this code is from an earlier time, at some point I read that you shouldn't use if/else at all. But yes, that should be good enough I don't forsee this code changing or getting more conditionals at any point.